tests/ice_frost.rs

```diff
@@ -272,3 +280,134 @@ fn signing_and_verification_3_out_of_5() {
+    let single_party_sk: SchnorrSecretKey = SchnorrSecretKey::rand(&mut rng);
+    let single_party_pk: SchnorrPublicKey =
+        <<Secp256k1Sha256 as CipherSuite>::G>::generator() * single_party_sk;
+
+    // Converts this party's keys into ICE-FROST format, simulating a 1-out-of-1 setup.
+    let simulated_parameters = ThresholdParameters::new(1, 1);
+    let frost_sk = IndividualSigningKey::from_single_key(single_party_sk);
+    let frost_pk = GroupVerifyingKey::new(single_party_pk);
+
+    // Start a resharing phase from this single party to a set of new participants.
+    const N_PARAM: u32 = 5;
+    const T_PARAM: u32 = 3;
+
+    let threshold_parameters = ThresholdParameters::new(N_PARAM, T_PARAM);
+
+    let mut signers = Vec::<Participant<Secp256k1Sha256>>::new();
+    let mut signers_dh_secret_keys = Vec::<DiffieHellmanPrivateKey<Secp256k1Sha256>>::new();
+
+    for i in 1..=N_PARAM {
+        let (p, dh_sk) =
+            Participant::<Secp256k1Sha256>::new_signer(threshold_parameters, i, rng).unwrap();
+
+        signers.push(p);
+        signers_dh_secret_keys.push(dh_sk);
+    }
+
+    let mut signers_encrypted_secret_shares: Vec<Vec<EncryptedSecretShare<Secp256k1Sha256>>> =
+        (0..N_PARAM).map(|_| Vec::new()).collect();
+
+    let mut signers_states_1 = Vec::<Dkg<_>>::new();
+    let mut signers_states_2 = Vec::<Dkg<_>>::new();
+
+    let (single_dealer, dealer_encrypted_shares_for_signers, _participant_lists) =
+        Participant::reshare(threshold_parameters, &frost_sk, &signers, rng).unwrap();
+
+    for i in 0..N_PARAM as usize {
+        let (signer_state, _participant_lists) =
+            DistributedKeyGeneration::<RoundOne, Secp256k1Sha256>::new(
+                simulated_parameters,
+                &signers_dh_secret_keys[i],
+                signers[i].index,
+                &[single_dealer.clone()],
+                rng,
+            )
+            .unwrap();
+        signers_states_1.push(signer_state);
+    }
+
+    for (i, shares) in signers_encrypted_secret_shares.iter_mut().enumerate() {
+        let share_for_signer = dealer_encrypted_shares_for_signers
+            .get(&(i as u32 + 1))
+            .unwrap()
+            .clone();
+        *shares = vec![share_for_signer];
+    }
+
+    for i in 0..N_PARAM as usize {
+        let (si_state, complaints) = signers_states_1[i]
+            .clone()
+            .to_round_two(&signers_encrypted_secret_shares[i], rng)
+            .unwrap();
+        assert!(complaints.is_empty());
+
+        signers_states_2.push(si_state);
+    }
+
+    let mut signers_secret_keys = Vec::<IndividualSigningKey<Secp256k1Sha256>>::new();
+
+    for signers_state in &signers_states_2 {
+        let (si_group_key, si_sk) = signers_state.clone().finish().unwrap();
+        signers_secret_keys.push(si_sk);
+
+        // Assert that each signer's individual group key matches the converted
+        // single's party public key.
+        assert!(si_group_key == frost_pk);
```

Comment:
So this proves that the input are the same, but can we also convert `frost_pk` back to the original `single_party_pk`? Do they have the same representation when serialized as bytes? If no, I wonder if that is a problem for topos validators?

Reply:
Yes they are purely identical from a byte perspective. The type difference is because one needs PhantomData to refer to the Frost ciphersuite.

Reply:
Do you think it's a good idea to toss in a `.to_bytes()` and compare that, as a way to communicate to readers that they're the same exact bytes?

Reply:
Public keys (i.e. elliptic curve points) are always compared on a byte-level when checking for equality (possibly with some canonical reduction, point compression techniques beforehand) so we can make it explicit and it would still work, but I'm not sure it brings much more than an explicit `key_a == key_b`.

Reply:
As long as validators do not need to change or convert their keys, I'm good with it.

Reply:
> I'm not sure it brings much more than an explicit key_a == key_b.

I guess it depends on how knowledgeable the reader is. :)